pkg/clients/cluster/cluster.go

```diff
@@ -49,8 +49,9 @@ const (
 )
 
 const (
-	errNoSecretInfo  = "missing secret information for GKE cluster"
-	errCheckUpToDate = "unable to determine if external resource is up to date"
+	errNoSecretInfo            = "missing secret information for GKE cluster"
+	errCheckUpToDate           = "unable to determine if external resource is up to date"
+	errAutopilotConfigMismatch = "autopilot configuration is different between observed and desired, unable to update"
```

Comment:
I think the following message is shorter and easier to understand:

```suggestion
	errAutopilotConfigMismatch = "autopilot configuration is immutable, unable to update"
```

Could you also add the `// +immutable` marker to the Autopilot field? Check [this ](https://github.com/crossplane-contrib/provider-gcp/blob/30ff5893b2a117ce20135a0edb5643ca896681a0/apis/cache/v1beta1/cloudmemorystore_instance_types.go#L30)example